searchlib/src/vespa/searchlib/attribute/multienumattribute.hpp

```diff
@@ -42,18 +44,31 @@ void
 MultiValueEnumAttribute<B, M>::reEnumerate()
 {
     // update MultiValueMapping with new EnumIndex values.
-    this->logEnumStoreEvent("compactfixup", "drain");
+    this->logEnumStoreEvent("compactfixup", "precompute");
+    vespalib::hash_map<EnumIndex, EnumIndex> old2new(this->_enumStore.getNumUniques());
```

Comment:
old2new should not be calculated here.
The mapping from old to new enum indexes should be filled out while performing the compaction in the enum store,
_indexMap should be removed from EnumStoreBase.

Reply:
Thanks, done. Please have another look.